backend/pkg/database/gorm/target.go

```diff
@@ -193,41 +217,62 @@ func (t *TargetsTableHandler) SaveTarget(target models.Target) (models.Target, e
 		return models.Target{}, fmt.Errorf("failed to convert API model to DB model: %w", err)
 	}
 
-	dbTarget.Data = marshaled
+	dbObj.Data = marshaled
 
-	if err = t.DB.Save(&dbTarget).Error; err != nil {
+	if err = t.DB.Save(&dbObj).Error; err != nil {
 		return models.Target{}, fmt.Errorf("failed to save target in db: %w", err)
 	}
 
 	// TODO(sambetts) Maybe this isn't required now because the DB isn't
 	// creating any of the data (like the ID) so we can just return the
 	// target pre-marshal above.
 	var apiTarget models.Target
-	if err = json.Unmarshal(dbTarget.Data, &apiTarget); err != nil {
+	if err = json.Unmarshal(dbObj.Data, &apiTarget); err != nil {
 		return models.Target{}, fmt.Errorf("failed to convert DB model to API model: %w", err)
 	}
 
 	return apiTarget, nil
 }
 
-func (t *TargetsTableHandler) UpdateTarget(target models.Target) (models.Target, error) {
+// nolint:cyclop
+func (t *TargetsTableHandler) UpdateTarget(target models.Target, params models.PatchTargetsTargetIDParams) (models.Target, error) {
 	if target.Id == nil || *target.Id == "" {
 		return models.Target{}, fmt.Errorf("ID is required to update target in DB")
 	}
 
-	var dbTarget Target
-	if err := getExistingObjByID(t.DB, targetSchemaName, *target.Id, &dbTarget); err != nil {
+	var dbObj Target
+	if err := getExistingObjByID(t.DB, targetSchemaName, *target.Id, &dbObj); err != nil {
 		return models.Target{}, err
 	}
 
 	var err error
-	dbTarget.Data, err = patchObject(dbTarget.Data, target)
+	var dbTarget models.Target
```

Comment:
need to probably rename target to asset in this PR :)

Reply:
or if this goes first, check again in the renaming PR

Reply:
I think this goes first as the renaming PR is blocked on other stuff